crates/starknet-types-core/src/felt/mod.rs

```diff
@@ -936,11 +938,23 @@ mod serde_impl {
             if serializer.is_human_readable() {
                 serializer.serialize_str(&format!("{:#x}", self))
             } else {
-                let mut seq = serializer.serialize_seq(Some(32))?;
-                for b in self.to_bytes_be() {
-                    seq.serialize_element(&b)?;
+                let bytes = self.to_bytes_be();
+                let first_significant_byte = bytes.iter().position(|&b| b != 0).unwrap_or(31);
+                if first_significant_byte > 1 {
+                    let len = 32 - first_significant_byte;
+                    let mut seq = serializer.serialize_seq(Some(len + 1))?;
```

Comment:
Because we are already storing the size. We should call `serialize_eq` with `None` in order to spare the extra byte which contains the len at the beginning.
Otherway there is no use to store the len a second time ourself

Reply:
It makes our worst case 33 bytes and our best 3 bytes (value, our len, their len). We want 32 and 2.